src/java/main/org/apache/zookeeper/server/NIOServerCnxn.java

```diff
@@ -694,7 +690,7 @@ public void sendResponse(ReplyHeader h, Record r, String tag) {
                 }
             }
          } catch(Exception e) {
-            LOG.warn("Unexpected exception. Destruction averted.", e);
+            throw new IOException(e);
```

Comment:
Just a suggestion: wdyt about bubbling up a more custom message with the IOException instead of just encapsulate the Exception? I mean, something like:

``
throw new IOException("sendMessage exception: blah blah", e);
``